api/pkg/webhooks/manager.go

```diff
@@ -0,0 +1,208 @@
+package webhooks
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+)
+
+type WebhookManager interface {
+	InvokeWebhooks(
+		context.Context,
+		EventType,
+		interface{},
+		func([]byte) error,
+		func(error) error,
+	) error
+}
+
+type SimpleWebhookManager struct {
+	WebhookClients map[EventType]map[WebhookType][]WebhookClient
```

Comment:
do we expect more `WebhookType` that are not async or sync? If not I prefer having the async and sync client define in separate fields, so we don't need to make the struct so nested. Also possibly no need for this type to exist.